src/crypto/CHIPCryptoPALOpenSSL.cpp

```diff
@@ -1560,5 +1562,233 @@ CHIP_ERROR Spake2p_P256_SHA256_HKDF_HMAC::PointIsValid(void * R)
     return error;
 }
 
+static void security_free_cert_list(X509_LIST * certs)
+{
+    if (certs)
+    {
+        sk_X509_pop_free(certs, X509_free);
+    }
+}
+
+CHIP_ERROR LoadCertsFromPKCS7(const uint8_t * pkcs7, X509DerCertificate * x509list, uint32_t * max_certs)
+{
+    CHIP_ERROR err    = CHIP_NO_ERROR;
+    X509_LIST * certs = NULL;
+    BIO * bio_cert    = NULL;
+    PKCS7 * p7        = NULL;
+    int p7_type       = 0;
+
+    VerifyOrExit(x509list != nullptr, err = CHIP_ERROR_INVALID_ARGUMENT);
+    VerifyOrExit(max_certs != nullptr, err = CHIP_ERROR_INVALID_ARGUMENT);
+
+    bio_cert = BIO_new_mem_buf(pkcs7, -1);
```

Comment:
-1 means null-terminated and uint8_t means binary.

Please either keep binary and then pass in a length parameter or make the pkcs7 "const char *" to make it clear we use null terminated strings.